third-party/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+include(WoofSettings)
+
+add_library(miniz STATIC miniz/miniz.c)
```

Comment:
The miniz upstream cmake file has a rule called "amalgamation" that can be used to create `miniz.c/.h` from its various source files.

Reply:
Then we have to configure and build a CMake subproject, I wanted to avoid this for simplicity. I think an exception can be made for miniz. Other libraries (cJSON, PFFFT) have single-file versions in the repo.

Reply:
cjson is one of those libraries which have perpared packages everywhere (Debian, MSYS2, brew, vcpkg, ...). So this leaves us with one project that we maintain as a submodule (spng) and another one which we carry around as released sources (miniz). Honestly, this sounds more chaotic that the current situation.

Regarding pffft, I understood that @ceski-1 wanted to check fttw3 as an alternative, so this would be another case of "common library expected everywhere".

Reply:
Yes, submodules overcomplicate things. Easy integration is one of the goals of creating these single-file libraries. Perhaps we should move all these dependencies to the `third_party` directory

> Regarding pffft, I understood that @ceski-1 wanted to check fttw3 as an alternative, so this would be another case of "common library expected everywhere".

PFFFT seems good, so why rewrite working code.

Reply:
> Yes, submodules overcomplicate things. Easy integration is one of the goals of creating these single-file libraries. Perhaps we should move them to the `third_party` directory

Yes, that's a good idea to keep things separate.

> PFFFT seems good, so why rewrite working code.

This would replace a simple solution that we would have to carry around with a well-maitained universal solution available everywhere. I don't say that this is a requirement for me or that pffft is bad in any way, just a thing to consider.